src/objects.rs

```diff
@@ -584,6 +606,23 @@ impl Time {
         self.0 % 60
     }
 }
+impl FromStr for Time {
+    type Err = TimeError;
+    fn from_str(time: &str) -> Result<Self, Self::Err> {
+        let mut t = time.split(':');
+        let (hours, minutes, seconds) = match (t.next(), t.next(), t.next(), t.next()) {
+            (Some(h), Some(m), Some(s), None) => (h, m, s),
+            _ => return Err(TimeError::WrongFormat),
+        };
+        let hours: u32 = hours.parse().map_err(|_err| TimeError::WrongFormat)?;
```

Comment:
You can `impl From<std::num::ParseFloatError> for TimeError` to automate these conversions.

Reply:
thanks for the hint, I added it using From<ParseIntError>